app/screens/global_threads/global_threads.tsx

```diff
@@ -36,15 +41,41 @@ const styles = StyleSheet.create({
     },
 });
 
-const GlobalThreads = ({componentId, globalThreadsTab}: Props) => {
+const testID = 'global_threads.threads_list';
+
+const GlobalThreads = ({componentId, globalThreadsTab, hasUnreads, teamId}: Props) => {
     const serverUrl = useServerUrl();
     const intl = useIntl();
     const switchingTeam = useTeamSwitch();
     const isTablet = useIsTablet();
 
+    const flatListRef = useRef<FlatList<ThreadModel>>(null);
+
     const defaultHeight = useDefaultHeaderHeight();
 
-    const [tab, setTab] = useState<GlobalThreadsTab>(globalThreadsTab);
+    const tabs = useMemo<Array<TabDefinition<GlobalThreadsTab>>>(() => [
+        {
+            name: defineMessage({
+                id: 'global_threads.allThreads',
+                defaultMessage: 'All your threads',
+            }),
+            id: 'all',
+            hasDot: false,
+        },
+        {
+            name: defineMessage({
+                id: 'global_threads.unreads',
+                defaultMessage: 'Unreads',
+            }),
+            id: 'unreads',
+            hasDot: hasUnreads,
+        },
+    ], [hasUnreads]);
+    const tabOnChange = useCallback(() => {
+        flatListRef.current?.scrollToOffset({offset: 0});
+    }, []);
+
+    const [tab, tabsProps] = useTabs<GlobalThreadsTab>(globalThreadsTab, tabs, tabOnChange);
```

Comment:
more of a question to understand this better. Should this be in the index.ts like withObservables and such?

Reply:
We use the index.ts to handle connection with the database. Nothing more. Why would you put this in the index.ts?

Reply:
in my head this is kind of a mixin that handles state across the app, same as the DB. I can totally see it there as well as here (and that happens for the database as well)
as I said, it is more about understanding the unspoken rules :)